bus-mapping/src/circuit_input_builder.rs

```diff
@@ -439,6 +439,8 @@ pub struct CallContext {
     /// call. It is incremented when a subcall in this call succeeds by the
     /// number of successful writes in the subcall.
     pub reversible_write_counter: usize,
+    /// Call data source (copy of caller's memory or tx input)
+    pub call_data_source: Vec<u8>,
```

Comment:
Just call it `call_data`?

```suggestion
    pub call_data: Vec<u8>,
```

Reply:
But it's not really the `call_data` we used to know, it's complete memory of caller instead of sliced. I was afraid it would be confusing so made it `call_dada_source` (then we can index the byte by absolute address instead of relative one). Not sure if it makes sense, I can also update this to the real `call_data`. What do you think?

Reply:
ah, I see. I thought it was just the real `call_data`. The reason you take a snapshot of the entire caller memory is that you want to use `calldata_offset + index` to read the caller memory in the `CALLDATALOAD`, right? I think it'd be fine

Reply:
Yeah, but I just think it's really confusing, so I made a change to have `call_data` instead of `call_data_source` in [`4d0ee80`](https://github.com/appliedzkp/zkevm-circuits/pull/434/commits/4d0ee80c3059effb56043b97071b72a8c5785fc1)